### ISSUE TITLE
Markdown - Allow Blank Lines In List

### ISSUE THREAD

**Issue by deckar01** [rouge-ruby/rouge#686]
**Source:**

```md
- Parent
    - Child

    - Distant child
```

**Output:**

```html
<span id="LC1" class="line" lang="markdown"><span class="p">-</span> Parent</span>
<span id="LC2" class="line" lang="markdown"><span class="p">    -</span> Child</span>
<span id="LC3" class="line" lang="markdown"></span>
<span id="LC4" class="line" lang="markdown"><span class="sb">    - Distant child</span></span>
```

**Preview:**

![screen shot 2017-06-06 at 8 26 21 am](https://cloud.githubusercontent.com/assets/3108007/26831531/3227aefa-4a92-11e7-8dc2-91759cb3fd55.png)

**Expected behavior:**

The "distant" list item should be styled as a list item, not code.

**References:**

All major markdown parsers treat the second list child as part of the list. [See the babelmark test](http://johnmacfarlane.net/babelmark2/?text=-+Parent%0A++++-+Child%0A%0A++++-+Distant+child) for more details.

**Comment by deckar01** [comment]
It looks like the issue is that the [indentation based code block](https://github.com/jneen/rouge/blob/db15bd0/lib/rouge/lexers/markdown.rb#L46) has higher precedence than the [list items](https://github.com/jneen/rouge/blob/master/lib/rouge/lexers/markdown.rb#L56-L60).

I suspect that the solution is to introduce a "list" state since list-like items take priority once a list item is encountered. The top priority would be finding a list item prefixed with an arbitrary number of newlines and spaces, then defer to the root behavior. 

The list state should allow:

 - Inline markdown formatting only
 - Continuation of list content with a single newline
 - Additional list items with arbitrary leading newlines and spaces
 - Returning to the root state only when:
   - One or more blank lines are followed by a non-list item

**Comment by miparnisari** [comment]
I think we can close this one.

![image](https://user-images.githubusercontent.com/5374887/58768237-e8210e80-854c-11e9-81d7-bcfe259b235b.png)

**Comment by pyrmont** [comment]
That looks broken in a different way. The text of each list item is appearing outside the relevant `<span>` tag which shouldn't be how it works.

**Comment by pyrmont** [comment]
I'm pretty sure this is still broken and should be fixed.

**Comment by pyrmont** [comment]
OK, I went and looked at this properly and @miparnisari supposition back in June of last year was correct. I was wrong (and wrong again when I wrote the message 19 hours ago!): this issue has been fixed. The mistake I made was that Rouge does not wrap plain text (such as the text in the list items) with HTML tags. The `-` characters are wrapped in HTML tags because they are marked as punctuation.

Sorry for the confusion everyone!
